core/src/main/java/nl/nn/adapterframework/doc/model/FrankDocModel.java

```diff
@@ -0,0 +1,213 @@
+package nl.nn.adapterframework.doc.model;
+
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import org.apache.logging.log4j.Logger;
+import org.springframework.core.annotation.AnnotationUtils;
+
+import lombok.Getter;
+import lombok.Setter;
+import nl.nn.adapterframework.doc.IbisDoc;
+import nl.nn.adapterframework.doc.IbisDocRef;
+import nl.nn.adapterframework.util.LogUtil;
+
+public class FrankDocModel {
+	private static final String JAVA_STRING = "java.lang.String";
+
+	private static Logger log = LogUtil.getLogger(FrankDocModel.class);
+
+	private @Getter List<FrankDocGroup> groups;
+	private @Getter Map<String, FrankElement> allElements = new HashMap<>();
+
+	public FrankDocModel() {
+		groups = new ArrayList<>();
+	}
+
+	public FrankElement findOrCreateFrankElement(Class<?> clazz) {
+		if(allElements.containsKey(clazz.getName())) {
+			return allElements.get(clazz.getName());
+		}
+		Class<?> superClass = clazz.getSuperclass();
+		FrankElement parent = superClass == null ? null : findOrCreateFrankElement(superClass);
+		FrankElement current = new FrankElement(clazz, parent);
+		current.setAttributes(createAttributes(clazz.getDeclaredMethods(), current));
+		allElements.put(current.getFullName(), current);
+		return current;
+	}
+
+	List<FrankAttribute> createAttributes(Method[] methods, FrankElement attributeOwner) {
+		Map<String, Method> setterAttributes = getAttributeToMethodMap(methods, "set");
+		Map<String, Method> getterAttributes = getGetterAndIsserAttributes(methods, attributeOwner);
+		Map<String, String> setterToAttributeName = new HashMap<>();
+		for(String attributeName: setterAttributes.keySet()) {
+			setterToAttributeName.put(setterAttributes.get(attributeName).getName(), attributeName);
+		}
+		List<FrankAttribute> result = new ArrayList<>();
+		for(Method method: methods) {
+			if(setterToAttributeName.containsKey(method.getName())) {
```

Comment:
waarom itereer je hier niet over setterAttributes?

Reply:
Ik heb commentaar toegevoegd om dat uit te leggen.

Reply:
Gedaan. Zoals het nu is. Kan je voor de zekerheid checken dat ik zodanig itereer dat de volgorde bewaard blijft?